site/content/en/concepts/packaging/_index.md

```diff
@@ -25,4 +25,54 @@ For compatibility with other arbitrary formats, kpt supports generating
 resource configuration data from templates, configuration DSLs, and programs
 using [source functions].
 
+## Nested Packages
+
+A `nested package` is a `kpt` package which contains 1 or more `kpt` packages
+in its directory tree.
+
+Here is an example directory structure of a `nested package` with name `foo`
+
+```sh
+foo # nested package
+├── Kptfile
+├── bar # nested package
+│   ├── Kptfile
+│   ├── baz # package
+│   │   ├── Kptfile
+│   │   └── cm.yaml
+│   └── deploy.yaml
+└── service.yaml
+```
+
+#### Use cases
+
+1. Package publishers need a way to pick and choose multiple component packages
+   which work well together, create a single `nested package` using them to
+   ship an out of the box application, maintain the package and abstract the
+   details from package consumers. Alternatively, package consumers may [update]
+   individual packages directly from the upstream sources.
+2. Package publishers need a way to create parameter values (ex: [setters]) to
+   be consistent across multiple component packages and make it easy for package
+   consumers to provide them with a single action.
+3. (Under development) Package consumers need a way to apply a set of component
+   packages in a single action to a live cluster while maintaining the ability
+   to manage them (ex: add/destroy) independently.
+
+#### Principles
+
+Here are the core principles of `nested packages`
+
+1. Each kpt package is an independent building block and should contain resources
+   (ex: setter definitions) in its `Kptfile`.
+2. If a package(ex: `bar`) is present in the directory tree of another package
```

Comment:
Not sure I follow #2. What do you mean by actions? Kpt functions act on resources of subpackage for example.

Reply:
Added that this is only supported for `cfg` commands currently.